src/wallet/wallet.cpp

```diff
@@ -4363,4 +4368,30 @@ bool CWallet::LoadActiveHDPubKey(const CExtPubKey& xpub, const std::optional<CKe
     // We should have had either a key or a crypted key. Having neither is an error.
     return Assume(false);
 }
+
+bool CWallet::SetActiveHDKey(const CExtKey& xprv)
+{
+    AssertLockHeld(cs_wallet);
+    std::optional<CKey> key;
+    std::vector<unsigned char> crypted_key;
+    if (IsCrypted()) {
+        if (IsLocked()) {
+            return false;
+        }
+        CKeyingMaterial secret(xprv.key.begin(), xprv.key.end());
+        if (!EncryptSecret(GetEncryptionKey(), secret, xprv.key.GetPubKey().GetHash(), crypted_key)) {
+            return false;
+        }
+    } else {
+        key = xprv.key;
+    }
+    return SetActiveHDKey(xprv.Neuter(), key, crypted_key);
+}
+
+bool CWallet::SetActiveHDKey(const CExtPubKey& xpub, const std::optional<CKey>& key, const std::vector<unsigned char>& crypted_key)
+{
+    AssertLockHeld(cs_wallet);
+    LoadActiveHDPubKey(xpub, key, crypted_key);
```

Comment:
In 2e2545c3a:

Missing check for `LoadActiveHDPubKey()` output.

Reply:
Done